[BUG] discover: panic occurs when using `--cluster-uri`

**Describe the bug**
When using `--cluster-uri` to specify a cluster URI via the command line during static discovery, [a panic occurs](https://github.com/OpenCHAMI/ochami/pull/52#issuecomment-3245745925).

```
$ ./ochami discover static -f yaml -d @nodes.yaml --discovery-version 1 --cluster-uri https://demo.openchami.cluster
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xa2c25f]

goroutine 1 [running]:
github.com/OpenCHAMI/ochami/cmd.getBaseURI(0x11c7fe0, {0xc087ff, 0x3})
        /home/exouser/ochami/cmd/lib.go:349 +0x43f
github.com/OpenCHAMI/ochami/cmd.getBaseURISMD(...)
        /home/exouser/ochami/cmd/lib.go:286
github.com/OpenCHAMI/ochami/cmd.init.func75(0x11c7fe0, {0xc08bbf?, 0x4?, 0xc08b4f?})
        /home/exouser/ochami/cmd/discover-static.go:56 +0x37
github.com/spf13/cobra.(*Command).execute(0x11c7fe0, {0xc0003ca600, 0x8, 0x8})
        /home/exouser/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1019 +0xa91
github.com/spf13/cobra.(*Command).ExecuteC(0x11c9320)
        /home/exouser/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x46f
github.com/spf13/cobra.(*Command).Execute(...)
        /home/exouser/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071
github.com/OpenCHAMI/ochami/cmd.Execute()
        /home/exouser/ochami/cmd/root.go:68 +0x1e
main.main()
        /home/exouser/ochami/main.go:8 +0xf
```

**To Reproduce**
Steps to reproduce the behavior:
1. Start with blank SMD and blank `ochami` config.
2. Create a `nodes.yaml`.
3. `ochami discover static -f yaml -d @nodes.yaml --cluster-uri <cluster_base_uri>`
4. Observe panic above.

**Expected behavior**
`--cluster-uri` should correctly pass the base URI to `ochami discover static` and not panic.


**Additional context**
The issue is occurring in this block:

https://github.com/OpenCHAMI/ochami/blob/caf429e9d1ab52f54269401c21a75c967fe62fb3/cmd/lib.go#L338-L352

--- a/cmd/discover/static/static.go
+++ b/cmd/discover/static/static.go
@@ -77,8 +77,7 @@ See ochami-discover(1) for more details.`,
 			}
 
 			// This endpoint requires authentication, so a token is needed
-			cli.SetToken(cmd)
-			cli.CheckToken(cmd)
+			cli.HandleToken(cmd)
 
 			// Create client to make request to SMD
 			smdClient, err := smd.NewClient(smdBaseURI, cli.Insecure)
@@ -575,6 +574,7 @@ See ochami-discover(1) for more details.`,
 	staticCmd.Flags().StringP("data", "d", "", "payload data or (if starting with @) file containing payload data (can be - to read from stdin)")
 	staticCmd.Flags().VarP(&cli.FormatInput, "format-input", "f", "format of input payload data (json,json-pretty,yaml)")
 	staticCmd.Flags().Bool("overwrite", false, "overwrite any existing information instead of failing")
+	staticCmd.Flags().String("uri", "", "absolute base URI or relative base path of SMD")
 
 	staticCmd.RegisterFlagCompletionFunc("format-input", cli.CompletionFormatData)
 	staticCmd.RegisterFlagCompletionFunc("discovery-version", cli.CompletionDiscoveryVersion)

--- a/internal/cli/cli.go
+++ b/internal/cli/cli.go
@@ -357,20 +357,25 @@ func GetBaseURI(cmd *cobra.Command, serviceName config.ServiceName) (string, err
 	// previously-set values while leaving unspecified ones alone.
 	if cmd.Flag("cluster-uri").Changed || (cmd.Flag("uri") != nil && cmd.Flag("uri").Changed) {
 		log.Logger.Debug().Msg("using base URI passed on command line")
-		ccc := config.ConfigClusterConfig{URI: cmd.Flag("cluster-uri").Value.String()}
-		switch serviceName {
-		case config.ServiceBoot:
-			ccc.BootService.URI = cmd.Flag("uri").Value.String()
-		case config.ServiceBSS:
-			ccc.BSS.URI = cmd.Flag("uri").Value.String()
-		case config.ServiceCloudInit:
-			ccc.CloudInit.URI = cmd.Flag("uri").Value.String()
-		case config.ServicePCS:
-			ccc.PCS.URI = cmd.Flag("uri").Value.String()
-		case config.ServiceSMD:
-			ccc.SMD.URI = cmd.Flag("uri").Value.String()
-		default:
-			return "", fmt.Errorf("unknown service %q specified when generating base URI", serviceName)
+		var ccc config.ConfigClusterConfig
+		if cmd.Flag("cluster-uri").Changed {
+			ccc.URI = cmd.Flag("cluster-uri").Value.String()
+		}
+		if cmd.Flag("uri") != nil && cmd.Flag("uri").Changed {
+			switch serviceName {
+			case config.ServiceBoot:
+				ccc.BootService.URI = cmd.Flag("uri").Value.String()
+			case config.ServiceBSS:
+				ccc.BSS.URI = cmd.Flag("uri").Value.String()
+			case config.ServiceCloudInit:
+				ccc.CloudInit.URI = cmd.Flag("uri").Value.String()
+			case config.ServicePCS:
+				ccc.PCS.URI = cmd.Flag("uri").Value.String()
+			case config.ServiceSMD:
+				ccc.SMD.URI = cmd.Flag("uri").Value.String()
+			default:
+				return "", fmt.Errorf("unknown service %q specified when generating base URI", serviceName)
+			}
 		}
 		clusterConfig = clusterConfig.MergeURIConfig(ccc)
 	}

--- a/pkg/client/client.go
+++ b/pkg/client/client.go
@@ -12,7 +12,6 @@ import (
 	"encoding/json"
 	"fmt"
 	"io"
-	"io/ioutil"
 	"net/http"
 	"net/url"
 	"os"
@@ -303,7 +302,7 @@ func (oc *OchamiClient) MakeRequest(method, uri string, headers *HTTPHeaders, bo
 		if resBodyLen > 0 {
 			var resBodyCopy bytes.Buffer
 			resBodyReader := io.TeeReader(res.Body, &resBodyCopy)
-			resBodyBytes, err := ioutil.ReadAll(resBodyReader)
+			resBodyBytes, err := io.ReadAll(resBodyReader)
 			if err != nil {
 				log.Logger.Error().Err(err).Msg("failed to read body for debug message")
 			}

--- a/pkg/client/client_test.go
+++ b/pkg/client/client_test.go
@@ -7,7 +7,6 @@ package client
 
 import (
 	"errors"
-	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
 	"os"
@@ -158,7 +157,7 @@ func TestFileToHTTPBody(t *testing.T) {
 	// Prepare a temp JSON file
 	dir := t.TempDir()
 	path := filepath.Join(dir, "payload.json")
-	if err := ioutil.WriteFile(path, []byte(`{"n":42}`), 0o644); err != nil {
+	if err := os.WriteFile(path, []byte(`{"n":42}`), 0o644); err != nil {
 		t.Fatalf("WriteFile: %v", err)
 	}
 
@@ -207,7 +206,7 @@ func TestReadPayload(t *testing.T) {
 	// Prepare a temp JSON file
 	dir := t.TempDir()
 	path := filepath.Join(dir, "data.json")
-	if err := ioutil.WriteFile(path, []byte(`{"k":7}`), 0o644); err != nil {
+	if err := os.WriteFile(path, []byte(`{"k":7}`), 0o644); err != nil {
 		t.Fatalf("WriteFile: %v", err)
 	}
 
@@ -280,19 +279,19 @@ func TestReadPayloadSlice(t *testing.T) {
 	// Prepare temp files
 	dir := t.TempDir()
 	jsonSinglePath := filepath.Join(dir, "single.json")
-	if err := ioutil.WriteFile(jsonSinglePath, []byte(`{"k":7}`), 0o644); err != nil {
+	if err := os.WriteFile(jsonSinglePath, []byte(`{"k":7}`), 0o644); err != nil {
 		t.Fatalf("WriteFile: %v", err)
 	}
 	jsonArrayPath := filepath.Join(dir, "array.json")
-	if err := ioutil.WriteFile(jsonArrayPath, []byte(`[{"k":1},{"k":2}]`), 0o644); err != nil {
+	if err := os.WriteFile(jsonArrayPath, []byte(`[{"k":1},{"k":2}]`), 0o644); err != nil {
 		t.Fatalf("WriteFile: %v", err)
 	}
 	yamlSinglePath := filepath.Join(dir, "single.yaml")
-	if err := ioutil.WriteFile(yamlSinglePath, []byte("k: 9\n"), 0o644); err != nil {
+	if err := os.WriteFile(yamlSinglePath, []byte("k: 9\n"), 0o644); err != nil {
 		t.Fatalf("WriteFile: %v", err)
 	}
 	yamlArrayPath := filepath.Join(dir, "array.yaml")
-	if err := ioutil.WriteFile(yamlArrayPath, []byte("- k: 3\n- k: 4\n"), 0o644); err != nil {
+	if err := os.WriteFile(yamlArrayPath, []byte("- k: 3\n- k: 4\n"), 0o644); err != nil {
 		t.Fatalf("WriteFile: %v", err)
 	}
 

